services/comm/app/operations/message_create.rb

```diff
@@ -24,6 +30,53 @@ def not_permitted(ctx, **)
     ctx[:errors].add(:user, 'not permitted to send message')
   end
 
+  def valid_recipient_and_phone_number(ctx, **)
+    ctx[:params][:recipient_id].present? || ctx[:params][:to].present?
+  end
+
+  def invalid_recipient_and_phone_number(ctx, **)
+    ctx[:errors].add(:recipient, 'or phone number is missing')
+  end
+
+  def match_recipient_and_phone_number(ctx, **)
+    params = ctx[:params]
```

Comment:
params should always be present, so you can make it part of the method signature:
`def match_recipient_and_phone_number(ctx, params:, **)`